bootstrap-salt.sh

```diff
@@ -1813,27 +1819,26 @@ __function_defined() {
 #                 a boot process, such as on AWS AMIs. This func will wait until the boot
 #                 process is finished so the script doesn't exit on a locked proc.
 #----------------------------------------------------------------------------------------------------------------------
+APT_ERR=$(mktemp /tmp/apt_error.XXXX)
 __wait_for_apt(){
     # Timeout set at 15 minutes
     WAIT_TIMEOUT=900
 
     # Run our passed in apt command
-    "${@}"
+    "${@}" 2>$APT_ERR
     APT_RETURN=$?
 
-    # If our exit code from apt is 100, then we're waiting on a lock
-    while [ $APT_RETURN -eq 100 ]; do
+    # Make sure we're not waiting on a lock
+    while [ $APT_RETURN -ne 0 ] && [ $(grep -c "^E: Could not get lock" $APT_ERR) -ge 1 ]; do
```

Comment:
```suggestion
    while [ $APT_RETURN -ne 0 ] && grep -q -v '^E: Could not get lock' "$APT_ERR"; do
```
@ripesensor We could make it little bit more clear by removing the command substitution.
If pattern not found, this is a success.

Reply:
Agreed.